update of `miniz_oxide` breaks rust 1.59

The newly released `v0.6.4` of [miniz_oxide](https://crates.io/crates/miniz_oxide/versions) breaks building with Rust `v1.59`.
`miniz_oxide` is a dependency of [flate2](https://crates.io/crates/flate2).

Running `cargo build` returns the followign error:
```
error[E0658]: use of unstable library feature 'int_abs_diff'
   --> /home/simone/.cargo/registry/src/github.com-1ecc6299db9ec823/miniz_oxide-0.6.4/src/inflate/core.rs:824:54
    |
824 |     if out_buf_size_mask == usize::MAX && source_pos.abs_diff(out_pos) == 1 && out_pos > source_pos {
    |                                                      ^^^^^^^^
    |
    = note: see issue #89492 <https://github.com/rust-lang/rust/issues/89492> for more information

error[E0658]: use of unstable library feature 'int_abs_diff'
   --> /home/simone/.cargo/registry/src/github.com-1ecc6299db9ec823/miniz_oxide-0.6.4/src/inflate/core.rs:833:61
    |
833 |     } else if out_buf_size_mask == usize::MAX && source_pos.abs_diff(out_pos) >= 4 && out_pos > source_pos {
    |                                                             ^^^^^^^^
    |
    = note: see issue #89492 <https://github.com/rust-lang/rust/issues/89492> for more information
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.5.1] - Unreleased
+### Fixed
+- Lock version of flate2 to support rust v1.59
+
 ## [0.5.0] - 2023-01-02
 ### Added
 - Initial Astarte Device SDK release

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -42,7 +42,7 @@ async-trait = "0.1.51"
 uuid = {version="1.2.2", features = ["v5", "v4"] }
 base64 = "0.21.0"
 webpki = "0.22.0"
-flate2 = "1.0"
+flate2 = "=1.0.24"
 
 [dev-dependencies]
 astarte-device-sdk-derive = {path = "./astarte-device-sdk-derive" }

